client/interface_wireguard.go

```diff
@@ -0,0 +1,90 @@
+package client
+
+import (
+	"github.com/go-routeros/routeros"
+)
+
+type InterfaceWireguard struct {
+	Id         string `mikrotik:".id"`
+	Name       string `mikrotik:"name"`
+	Comment    string `mikrotik:"comment"`
+	Disabled   bool   `mikrotik:"disabled"`
+	ListenPort int    `mikrotik:"listen-port"`
+	Mtu        int    `mikrotik:"mtu"`
+	PrivateKey string `mikrotik:"private-key"`
+	PublicKey  string `mikrotik:"public-key"`       //read only property
```

Comment:
Ah good catch, I'll add the readonly tag to public key as well. I believe the test was passing because PublicKey defaults to an empty string so it didn't show up when adding a new interface wireguard. However since Running defaults to false, the test was complaining that I was trying to create an interface wireguard with a parameter that shouldn't exist (below).

![Screen Shot 2023-05-12 at 11 05 46 AM](https://github.com/ddelnano/terraform-provider-mikrotik/assets/64564343/f823c8ca-4dd7-4a4f-8adf-4c89e25b35f3)